.github/scripts/check-anchors.py

```
@@ -0,0 +1,343 @@
+        files.extend(glob.glob(os.path.join(root, pat), recursive=True))
+    return [f for f in files
+            if not any(x in f for x in ('node_modules/', '.git/', '.github/', 'tmp/'))]
+
+
+def git_changed_files(root: str) -> list:
+    """Files changed vs origin/main (fallback: HEAD~1 for direct pushes).
+
+    Returns both sides of a rename (pre- and post-image paths) so inbound
+    links to the old path are still validated after a file move.
+    """
+    out = subprocess.run(
```

Ruff (0.16.1)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)